psl-core/src/main/java/org/linqs/psl/util/VizDataCollection.java

```diff
@@ -0,0 +1,233 @@
+package org.linqs.psl.util;
+
+import org.linqs.psl.grounding.GroundRuleStore;
+import org.linqs.psl.model.atom.Atom;
+import org.linqs.psl.model.atom.GroundAtom;
+import org.linqs.psl.model.rule.GroundRule;
+import org.linqs.psl.model.rule.Rule;
+import org.linqs.psl.model.rule.UnweightedGroundRule;
+import org.linqs.psl.model.rule.WeightedGroundRule;
+import org.linqs.psl.model.rule.logical.AbstractLogicalRule;
+import org.linqs.psl.model.term.Constant;
+import org.linqs.psl.model.term.Variable;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileWriter;
+import java.util.HashMap;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.HashSet;
+
+public class VizDataCollection {
+
+    private static VizDataCollection vizData = null;
+    private static Runtime runtime = null;
+
+    private JSONObject fullJSON;
+    private JSONArray predictionTruthArray;
+    private JSONArray ruleCountArray;
+    private JSONArray totRuleSatArray;
+    private JSONArray violatedGroundRulesArray;
+
+    JSONObject rules;
+    JSONObject groundRules;
+    JSONObject groundAtoms;
+
+    public static ArrayList<GroundRule> violatedGroundRulesList = new ArrayList<>();
```

Comment:
This should be private and non-static.

Even if violated ground rules were computed here (which will be moved to the UI), this is not the correct way to do it.
Follow the same pattern as the rest of the data.

Reply:
Resolved